### PR TITLE
Enable branch protection and auto merge

### PR DESCRIPTION
**Pull request by laeubi**
This enables branch protection and automerge

### CODE REVIEW

File: otterdog/eclipse-tycho.jsonnet
Comment (laeubi) [inline otterdog/eclipse-tycho.jsonnet]: I'm not totally sure about this, what I found in some examples it seem to need match a name.

Basically I want that this check is mandatory:
https://github.com/eclipse-tycho/tycho/actions/workflows/licensecheck.yml

Comment (netomi) [reply otterdog/eclipse-tycho.jsonnet]: the additional eclipsefdn/eca check is is not needed, as it is already inherited from the default.

For the license checking status, you should use

"call-license-check / check-licenses"

I checked it via the web ui and that the status that is detected. It follows the rules that are explained here: https://otterdog.readthedocs.io/en/latest/reference/organization/repository/status-check/

the jenkins status is correct afacit.

Comment (netomi) [reply otterdog/eclipse-tycho.jsonnet]: here is a screenshot from the web ui:

image

Comment (laeubi) [reply otterdog/eclipse-tycho.jsonnet]: Is there a link to the webui? Is it possible to edit the project file with that so one don't need to edit file by hand?

**Comment by laeubi** [comment]
@netomi thanks for the hints, I now updated the PR.

**Comment by netomi** [comment]
LGTM, can some project lead approve that change?

**Comment by laeubi** [comment]
I approve this as PL of Tycho project, maybe @akurtakov can give approval as well.

**Review by akurtakov — APPROVED** [review]
Fine with me.

**Comment by laeubi** [comment]
@netomi one thing I noticed:

I cant approve this PR because it says 

> Pull request authors can’t approve their own pull request 

that seems a bit odd here as a PL can then never approve its own proposed changes :-D

**Comment by netomi** [comment]
> @netomi one thing I noticed:
> 
> I cant approve this PR because it says
> 
> > Pull request authors can’t approve their own pull request
> 
> that seems a bit odd here as a PL can then never approve its own proposed changes :-D

thats normal GitHub behavior. You cant approve your own PRs.

**Review by netomi — APPROVED** [review]

**Comment by netomi** [comment]
changes are live. Two minor things that I had to fix:

- topics must start with a lowercase letter or number, so OSGi was not allowed, I changed that to osgi, will add a validation rule
- the committers team could not be used as bypass actor as it is a secret team, and secret teams can not be used for that. We plan to change the team to public but for now I enabled bypass for all actors with write permissions to the repo which is more or less the same as committers

**Comment by laeubi** [comment]
@netomi thanks, anyone with write access is fine, we want to use the protection rules manly for auto merge feature and otherwhise trust that committers do the right things (e.g. only bypass checks if they are sure it will fix something very important and the check is broken).

**Pull request merged**
